hack/tools/go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/metal3-io/cluster-api-provider-metal3/hack/tools
 
-go 1.23
+go 1.23.0
```

Comment:
And now they're both back by dependabot.

Reply:
Also the `.0` is back.

Reply:
`go mod tidy` will automatically add the patch version `.0`. And if we don't run it, `generate` test will fail.

Reply:
So it seems. We have many that do not have the .0, but in this one it really does want it there for some reason.